queue-worker/main.go

```diff
@@ -90,11 +90,15 @@ func main() {
 		req := queue.Request{}
 		json.Unmarshal(msg.Data, &req)
 		fmt.Printf("Request for %s.\n", req.Function)
-		urlFunction := fmt.Sprintf("http://%s%s:8080/", req.Function, functionSuffix)
+		functionURL := fmt.Sprintf("http://%s%s:8080/?%s", req.Function, functionSuffix, req.QueryString)
 
-		request, err := http.NewRequest("POST", urlFunction, bytes.NewReader(req.Body))
+		request, err := http.NewRequest("POST", functionURL, bytes.NewReader(req.Body))
 		defer request.Body.Close()
 
+		for k, v := range req.Header {
```

Comment:
Should we check to see we're not overwriting anything?

Reply:
I thought about it for a sec, but I think we should stick to the original headers. Otherwise we are sort of altering the original request.